confidential-datahub/kms/src/api.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! # Api definitions for KMS/Vault drivers
+//!
+//! To establish a connection between the client and the KMS/Vault server, two
+//! kinds of parameters are required:
+//! - Public parameters: like Region Id of the KMS/Vault, Instance Id of the KMS,
+//! etc. They are not confidential and can be seen by anyone. [`ProviderSetting`]
+//! is a json object. It is to include all the public parameters. The
+//! hashmap style makes it flexible for different kinds of KMSes/Vaults. There
+//! should be a function that takes a [`ProviderSetting`] as a input parameter
+//! and returns a client instance.
+//! - Private parameters: like the credential to access (e.g. access key).
```

Comment:
What about the URI of a KMS? That isn't private, but it should be provided via new, right?

Reply:
Yes! The URI of a KMS is a "public parameter", so it should be put in `provider_settings`. Taking aliyun as example, the URL of the KMS is derived by the `kms_instance_id`, so we do not directly have URI of a KMS to `new()`, but derive it inside `new()`